### PR TITLE
Additional release notes for Keycloak 24

### LINKED ISSUE #27142: Add additional contributions to the release notes
### Description

This is to not forget about adding additional contributions to the release notes. GH filter for things, which should be added: https://github.com/keycloak/keycloak/issues?q=is%3Aissue+is%3Aclosed+label%3Ateam%2Fcore-clients+label%3Arelease%2F24.0.0+label%3Arelease-notes+closed%3A%3E2024-02-18+

NOTE:
- PR for secure-redirect uri executor was co-authored by https://github.com/lexcao and he should be added to the release notes for this one together with Takashi.

- PR for OAuth 2.1 was co-authored by https://github.com/skabano .

### PR DESCRIPTION
**Pull request by mposolda**
closes #27142

### CODE REVIEW

File: docs/documentation/release_notes/topics/24_0_0.adoc [hunk 1]
```diff
@@ -130,23 +130,46 @@ This release contains support for Lightweight access tokens. As a result, you ca
...
+
 ifeval::[{project_community}==true]
 Thanks to https://github.com/skabano[Shigeyuki Kabano] for the contribution and Thanks to
 https://github.com/tnorimat[Takashi Norimatsu] for a help and review of this feature.
 endif::[]
 
+== OAuth 2.1 support
+
+This release contains optional OAuth 2.1 support. New client policy profiles were introduced in this release, which administrator can use to make sure that his clients and particular client requests
+will be compliant with OAuth 2.1 specification. There is dedicated client profile for confidential clients and dedicated profile for public clients.
```
Comment (andymunro) [inline docs/documentation/release_notes/topics/24_0_0.adoc:151]:
```suggestion
This release contains optional OAuth 2.1 support. New client policy profiles were introduced in this release, which administrators can use to make sure that clients and particular client requests comply with the OAuth 2.1 specification. A dedicated client profile exists for confidential clients and a dedicated profile for public clients.
```

File: docs/documentation/release_notes/topics/24_0_0.adoc [hunk 2]
```diff
@@ -130,23 +130,46 @@ This release contains support for Lightweight access tokens. As a result, you ca
...
+Starting with this release, the *scope* parameter in the OAuth2/OIDC endpoint for token refresh is supported. Use this parameter to request access tokens with a smaller amount
 of scopes than originally granted, which means you cannot increase access token scope. This scope limitation does not affect the scope of the refreshed refresh token. This function works as
 described in the OAuth2 specification.
 ifeval::[{project_community}==true]
 Thanks to https://github.com/cgeorgilakis[Konstantinos Georgilakis] for the contribution.
 endif::[]
 
+== Client policy executor for secure redirect URIs
+
+A new client policy executor `secure-redirect-uris-enforcer` is introduced. Use it to restrict what are the redirect URIs, which can be used by the clients. For instance
```
Comment (andymunro) [inline docs/documentation/release_notes/topics/24_0_0.adoc:167]:
```suggestion
A new client policy executor `secure-redirect-uris-enforcer` is introduced. Use it to restrict which redirect URIs can be used by the clients. For instance, 
```

File: docs/documentation/release_notes/topics/24_0_0.adoc [hunk 3]
```diff
@@ -172,6 +202,14 @@ ifeval::[{project_community}==true]
 Thanks to https://github.com/MikeTangoEcho[MT] for the contribution.
 endif::[]
 
+== OAuth Grant Type SPI
+
+Internal update in the {project_name} codebase was done to introduce OAuth Grant Type SPI. This allows additional flexibility when introducing custom grant types
```
Comment (andymunro) [inline docs/documentation/release_notes/topics/24_0_0.adoc:207]:
```suggestion
The {project_name} codebase includes an internal update  to introduce the OAuth Grant Type SPI. This update allows additional flexibility when introducing custom grant types
```

File: docs/documentation/release_notes/topics/24_0_0.adoc [hunk 4]
```diff
@@ -172,6 +202,14 @@ ifeval::[{project_community}==true]
...
+supported by {project_name} OAuth 2 token endpoint.
```
Comment (andymunro) [inline docs/documentation/release_notes/topics/24_0_0.adoc:208]:
```suggestion
supported by the {project_name} OAuth 2 token endpoint.
```

File: docs/documentation/server_admin/topics/clients/client-policies.adoc [hunk 5]
```diff
@@ -79,6 +79,10 @@ Client Role::
     Applies for clients with the client role of the specified name. Typically you can create a client role of specified name to requested clients and use it as a "marker role" to make
     sure that specified client policy will be applied for requested clients.
 
+NOTE: There is often a use-case for require that particular client policy is applied for the specified clients like `my-client-1` and `my-client-2`. The best way to achieve this
+is to use *Client Role* condition in your policy and then create client role of specified name to requested clients. This client role can be used as a "marker role" used solely
+for marking that particular client policy should be used for particular clients.
```
Comment (andymunro) [inline docs/documentation/server_admin/topics/clients/client-policies.adoc:84]:
```suggestion
NOTE: A use-case often exists for requiring the application of a particular client policy for the specified clients such as `my-client-1` and `my-client-2`. The best way to achieve this result 
is to use a *Client Role* condition in your policy and then a create client role of specified name to requested clients. This client role can be used as a "marker role" used solely
for marking that particular client policy should be used for particular clients.
```

**Review by andymunro — CHANGES_REQUESTED** [review]
Looks good, @mposolda. A few suggestions.

**Comment by mposolda** [comment]
@andymunro Thanks!

**Pull request merged**